rust/kernel/net/skbuff.rs

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Definitors for the struct [`crate::bindings::sk_buff`] memory handlers.
+
+use core::{ops::Drop, ptr};
+
+use crate::bindings;
+use crate::types::SavedAsPointer;
+
+/// Wraps the kernel's `struct sk_buff`.
+///
+/// # Invariants
+///
+/// The pointer `Self::ptr` is non-null and valid.
+#[repr(transparent)]
+pub struct SkBuff {
+    ptr: *const bindings::sk_buff,
+}
+
+impl SkBuff {
+    /// Binding to `bindings::skb_clone_tx_timestamp`.
+    ///
+    /// Noop if `CONFIG_NETWORK_PHY_TIMESTAMPING` is not enabled.
+    #[cfg(CONFIG_NETWORK_PHY_TIMESTAMPING)]
+    pub fn clone_tx_timestamp(&mut self) {
+        // SAFETY: self.ptr is valid if self is valid.
+        unsafe {
+            bindings::skb_clone_tx_timestamp(self.ptr as *mut bindings::sk_buff);
+        }
+    }
+
+    /// Binding to `bindings::skb_clone_tx_timestamp`.
+    ///
+    /// Noop if `CONFIG_NETWORK_PHY_TIMESTAMPING` is not enabled.
```

Comment:
Since here we know it is not enabled, we can explain it is a noop.

```suggestion
    /// Noop since `CONFIG_NETWORK_PHY_TIMESTAMPING` is disabled.
```